python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -37,6 +38,19 @@
 __all__: list[str] = ["Column"]
 
 
+def _dtype_short_repr_to_dtype(dtype_str: str) -> pl.DataType:
+    """Convert a Polars dtype short repr to a Polars dtype."""
+    # limitations of dtype_short_repr_to_dtype described in
+    # py-polars/polars/datatypes/convert.py#L299
+    if dtype_str.startswith("list["):
+        stripped = dtype_str.removeprefix("list[").removesuffix("]")
+        return pl.List(_dtype_short_repr_to_dtype(stripped))
+    pl_type = pl.datatypes.convert.dtype_short_repr_to_dtype(dtype_str)
+    if pl_type is None:
+        raise ValueError(f"{dtype_str} was not able to be parsed by Polars.")
+    return pl_type() if inspect.isclass(pl_type) else pl_type
```

Comment:
How safe is `pl_type()`, without any arguments, here? Some types (`Array`, `Enum`) require additional arguments. Maybe we don't support those yet?

Reply:
For the types we support in `DataType`, I believe this is fairly safe as I'm hoping that `dtype_short_repr_to_dtype` will return instances for types with parameters (`polars.Datetime` and `polars.Duration`).

For those types that we don't support that take arguments, those should be rejected when constructing a `DataType`